### PR TITLE
Implementation and testing of config/cli.go 

### LINKED ISSUE #15: Implement `config/cli.go`

### PR DESCRIPTION
**Pull request by ABD-AZE**
Resolves #15 
1) Building and testing done successfully
2) CliConfig Struct defined
3) as_provider() function serializes the struct to map[string]{interface}
4) written test for as_provider() function using viper for testing but any other config management lib can be used for the project as the testing was only for checking the serialization process.

### CODE REVIEW

**Comment by star-gazer111** [comment]
@ABD-AZE We are using Viper for now right?

**Comment by star-gazer111** [comment]
LGTM

**Comment by ABD-AZE** [comment]
I have used viper just for testing that the data is getting serialised. It is not necessary to use viper for the whole project.

**Review by star-gazer111 — APPROVED** [review]

File: config/cli.go [hunk 1]
```diff
@@ -1,8 +1,50 @@
 package config
 
+import (
+    "encoding/hex"
+)
+// The format of configuration to be stored in the configuratin file is map[string]interface{}
 type CliConfig struct {
+    ExecutionRPC         *string `mapstructure:"execution_rpc"`
```
Comment (gerceboss) [inline config/cli.go:8]: Change `ExecutionRPC` to `ExecutionRpc`, similarly do for RPC and IP everywhere

**Pull request merged**